app/models/visitor_supersession.rb

```diff
@@ -6,15 +6,24 @@ class VisitorSupersession < ActiveRecord::Base
 
   private
 
-  def merge_assignments! # rubocop:disable Metrics/AbcSize
+  def merge_assignments!
     target_split_ids = superseding_visitor.assignments.map(&:split_id).to_set
-    superseded_visitor.assignments.order(:id).map do |a|
+    superseded_visitor.assignments.order(:id).each do |a|
+      create_or_ignore_duplicate(a) unless target_split_ids.include?(a.split_id)
+    end
+  end
+
+  def create_or_ignore_duplicate(assignment)
+    transaction(requires_new: true) do
       superseding_visitor.assignments.create!(
-        variant: a.variant,
-        split_id: a.split_id,
+        variant: assignment.variant,
+        split_id: assignment.split_id,
         context: 'visitor_supersession',
         visitor_supersession: self
-      ) unless target_split_ids.include?(a.split_id)
+      )
     end
+  rescue ActiveRecord::RecordNotUnique, PG::UniqueViolation # rubocop:disable Lint/HandleExceptions
+    # ignore or find
+    # we're not using the result, so probably just ignore
```

Comment:
I'd say why we can ignore - the goal is to fill gaps, not override anything, or read anything.